maps/src/main/java/org/odk/collect/maps/layers/OfflineMapLayersPicker.kt

```diff
@@ -120,4 +139,60 @@ class OfflineMapLayersPicker(
             // ignore
         }
     }
+
+    override fun onLayerChecked(layerId: String?) {
+        stateViewModel.onLayerChecked(layerId)
+    }
+
+    override fun onLayerToggled(layerId: String?) {
+        stateViewModel.onLayerToggled(layerId)
+    }
+
+    override fun onDeleteLayer(layerItem: CheckableReferenceLayer) {
+        MaterialAlertDialogBuilder(requireActivity())
+            .setMessage(requireActivity().getLocalizedString(org.odk.collect.strings.R.string.delete_layer_confirmation_message, layerItem.name))
+            .setPositiveButton(org.odk.collect.strings.R.string.delete_layer) { _, _ ->
+                layerItem.file?.delete()
```

Comment:
Ok, I've moved deleting to the view model but I think that the whole list should not be loaded from scratch when we delete one layer. This would be a bad user experience. Updating the list of layers is faster.